processors/blockprocessor/block_processor.go

```diff
@@ -27,8 +29,112 @@ type blockProcessor struct {
 	logger  *log.Logger
 }
 
-// MakeProcessorWithLedger creates a block processor with a given ledger
-func MakeProcessorWithLedger(logger *log.Logger, l *ledger.Ledger, handler func(block *ledgercore.ValidatedBlock) error) (processor.Processor, error) {
+// Constructor is the ProcessorConstructor implementation for the "block processor" processor
+type Constructor struct{}
+
+// New initializes a postgresqlExporter
+func (c *Constructor) New() processors.Processor {
+	return &blockProcessor{}
+}
+
+var implName string = "Block Processor"
+
+func (proc *blockProcessor) Metadata() processors.ProcessorMetadata {
+	return processors.ProcessorMetadata{
+		ImpName:        implName,
+		ImpDescription: "Local Ledger Block Processor",
+		ImpDeprecated:  false,
+	}
+}
+
+func (proc *blockProcessor) Config() plugins.PluginConfig {
+	return ""
+}
+
+func (proc *blockProcessor) Init(ctx context.Context, cfg plugins.PluginConfig) error {
+	processors.RegisterProcessor(implName, &Constructor{})
+	return nil
+}
+
+func (proc *blockProcessor) Close() error {
+	return nil
```

Comment:
Can also include cancellation of a local context.Context variable initialized when Init(...) was called

Reply:
@algoganesh isn't the context passed from the Conduit framework?  Wouldn't that mean that Conduit would need to cancel it?

Reply:
I was considering situations when the framework would like to abort ongoing actions by a particular plugin by calling that plugin's `Close()` function without having to cancel the original context, since that original context could be mutual to other plugins.